### PR TITLE
Add trigger rumble effect

### LINKED ISSUE #138: Xbox One impulse trigger effects
The haptic vibration proposal (https://github.com/w3c/gamepad/pull/68) only defines the typical "dual-rumble" style rumble effect with strong/weak ERM actuators. Xbox One controllers additionally have [Impulse Triggers](https://www.pcgamingwiki.com/wiki/Controller:Xbox_One_Controller#Impulse_Trigger_Vibration): haptic actuators that can be used to create effects localized to a particular trigger. To support these let's add new actuator types, for instance:

        enum GamepadHapticActuatorType {	
          "dual-rumble",
          "impulse-trigger-left",  // NEW
          "impulse-trigger-right",  // NEW
        };

Xbox One impulse triggers can be controlled on Windows 10 using the [Windows.Gaming.Input API](https://docs.microsoft.com/en-us/uwp/api/windows.gaming.input.gamepadvibration?view=winrt-19041).

### PR DESCRIPTION
**Pull request by gabrielsanbrito**
Closes #138

The following tasks have been completed:

 * [ ] Modified Web platform tests (link to pull request)

Implementation commitment:

 * [X] [WebKit](https://bugs.webkit.org/show_bug.cgi?id=250352) - implemented - not shipping (pref’ed off)
 * [X] [Chromium](https://issues.chromium.org/issues/40834175) - implemented and shipping
 * [ ] Gecko (https://bugzilla.mozilla.org/show_bug.cgi?id=1824217)

***
<a href="https://pr-preview.s3.amazonaws.com/gabrielsanbrito/gamepad/pull/195.html" title="Last updated on Mar 29, 2024, 11:48 PM UTC (e58147a)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/gamepad/195/c207166...gabrielsanbrito:e58147a.html" title="Last updated on Mar 29, 2024, 11:48 PM UTC (e58147a)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1455,12 +1456,54 @@ <h3>
...
         </dd>
+        <dt>
+          <dfn data-for="GamepadHapticEffectType">"trigger-rumble" effect
+          type</dfn>
+        </dt>
+        <dd>
+          <p>
+            {{GamepadHapticEffectType/"trigger-rumble"}} describes a haptics
+            configuration with a vibration motor in each of the bottom front buttons
+            of a [=Standard Gamepad=] (buttons with [=canonical indexes=] 6 and 7) in addition to the two handle motors used for
```
Comment (nondebug) [inline index.html]: Both "indexes" and "indices" are valid plurals for "index", but we should choose one and use it consistently within the spec. "Indices" is already used elsewhere in the spec so I think we should prefer `[=canonical indices=]` here.

File: index.html [hunk 2]
```diff
@@ -1455,12 +1456,54 @@ <h3>
...
+          <dfn data-for="GamepadHapticEffectType">"trigger-rumble" effect
+          type</dfn>
+        </dt>
+        <dd>
+          <p>
+            {{GamepadHapticEffectType/"trigger-rumble"}} describes a haptics
+            configuration with a vibration motor in each of the bottom front buttons
+            of a [=Standard Gamepad=] (buttons with [=canonical indexes=] 6 and 7) in addition to the two handle motors used for 
+            {{GamepadHapticEffectType/"dual-rumble"}}. Nowadays these buttons
+            most-commonly take the form of spring-loaded triggers. In this
```
Comment (nondebug) [inline index.html]: "most commonly" (no dash)

File: index.html [hunk 3]
```diff
@@ -1455,12 +1456,54 @@ <h3>
...
+            {{GamepadEffectParameters/strongMagnitude}},
+            {{GamepadEffectParameters/weakMagnitude}} share the same definition with {{GamepadHapticEffectType/"dual-rumble"}}.
+            {{GamepadEffectParameters/leftTrigger}} and
+            {{GamepadEffectParameters/rightTrigger}}, respectively, set the
+            intensity levels for the left and right bottom front buttons
+            vibrations, normalized to the range `[0,1]`, defaulting to 0.
+          </p>
+          <p>
+            Given {{GamepadEffectParameters}} |params:GamepadEffectParameters|,
+            a <dfn>valid trigger-rumble effect</dfn> must have a valid
```
Comment (nondebug) [inline index.html]: Please update the definition of "valid effect" to also check for valid trigger-rumble effects. (Currently it only checks dual-rumble and will always consider trigger-rumble effects valid.)

Comment (gabrielsanbrito) [reply index.html]: Thanks for pointing it out!

**Review by nondebug — APPROVED** [review]
lgtm

File: index.html [hunk 4]
```diff
@@ -1455,12 +1463,54 @@ <h3>
...
+        <dt>
+          <dfn data-for="GamepadHapticEffectType">"trigger-rumble" effect
+          type</dfn>
+        </dt>
+        <dd>
+          <p>
+            {{GamepadHapticEffectType/"trigger-rumble"}} describes a haptics
+            configuration with a vibration motor in each of the bottom front buttons
+            of a [=Standard Gamepad=] (buttons with [=canonical indices=] 6 and 7) in addition to the two handle motors used for 
+            {{GamepadHapticEffectType/"dual-rumble"}}. Nowadays these buttons
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
            {{GamepadHapticEffectType/"dual-rumble"}}. These buttons
```

File: index.html (hunk 3 above)
Comment (marcoscaceres) [reply index.html]: @gabrielsanbrito, can you link to "valid effect" where "valid" is mentioned in this paragraph, I think `[=valid effect|valid=]` should work...

**Review by marcoscaceres — APPROVED** [review]
Couple of small comments. Thanks again @gabrielsanbrito!

**Comment by marcoscaceres** [comment]
Fixed link to webkit implementation bug too.

File: index.html (hunk 3 above)
Comment (gabrielsanbrito) [reply index.html]: @marcoscaceres like below? Should I also add it inside the `dfn` tag? `<dfn>[=valid effect|valid=] trigger-rumble effect</dfn>`
```html
Given {{GamepadEffectParameters}} |params:GamepadEffectParameters|,
a <dfn>[=valid effect|valid=] trigger-rumble effect</dfn> must have a [=valid effect|valid=]
{{GamepadEffectParameters/duration}}, a [=valid effect|valid=] {{GamepadEffectParameters/startDelay}}, and the
{{GamepadEffectParameters/strongMagnitude}},
{{GamepadEffectParameters/weakMagnitude}}, {{GamepadEffectParameters/leftTrigger}}, and {{GamepadEffectParameters/rightTrigger}} must be in the range `[0,1]`.
```

btw, `[=valid effect|valid=]` works.

Comment (marcoscaceres) [reply index.html]: No. The dfn is fine as it is.

Comment (gabrielsanbrito) [reply index.html]: Awesome! Did the same thing for the `dual-rumble` section too. Thanks!

**Comment by gabrielsanbrito** [comment]
@marcoscaceres I have addressed the last piece of feedback. However, I can't merge the PR :/ Could you please do it?

**Pull request merged**